### PR TITLE
Integration of HAL_GPIO

### PR DESCRIPTION
**Pull request by mariuscrsn**

### CODE REVIEW

File: firmware/Sources/HAL/HAL_GPIO/hal_gpio.c [hunk 1]
```diff
@@ -0,0 +1,132 @@
+/*********************************************************************************
+* @file           : hal_gpio.c
+* @brief          : Implementation of HAL GPIO
+**********************************************************************************
+* @attention
```
Comment (pastorpflores) [inline firmware/Sources/HAL/HAL_GPIO/hal_gpio.c:5]: delete this header

File: firmware/Sources/HAL/HAL_GPIO/hal_gpio_test.c [hunk 2]
```diff
@@ -0,0 +1,91 @@
+/*
+ * TestingGpio.c
+ *
+ *  Created on: 25 may. 2023
```
Comment (pastorpflores) [inline firmware/Sources/HAL/HAL_GPIO/hal_gpio_test.c:4]: update this header to the standard

File: firmware/Sources/HAL/HAL_GPIO/hal_gpio_test.h [hunk 3]
```diff
@@ -0,0 +1,43 @@
+/*
+ * TestingGpio.h
+ *
+ *  Created on: 25 may. 2023
```
Comment (pastorpflores) [inline firmware/Sources/HAL/HAL_GPIO/hal_gpio_test.h:4]: update this header to the standard

File: firmware/Sources/Test/HAL/hal_gpio_test.c [hunk 4]
```diff
@@ -0,0 +1,98 @@
+/*
+ * TestingGpio.c
+ *
+ *  Created on: 25 may. 2023
```
Comment (pastorpflores) [inline firmware/Sources/Test/HAL/hal_gpio_test.c:4]: update this header to the standard

File: firmware/Sources/Test/HAL/hal_gpio_test.h [hunk 5]
```diff
@@ -0,0 +1,43 @@
+/*
+ * TestingGpio.h
+ *
+ *  Created on: 25 may. 2023
```
Comment (pastorpflores) [inline firmware/Sources/Test/HAL/hal_gpio_test.h:4]: update this header to the standard

**Review by pastorpflores — CHANGES_REQUESTED** [review]

File: firmware/Sources/HAL/HAL_GPIO/hal_gpio_test.h (hunk 3 above)
Comment (RobertoAldea) [reply firmware/Sources/HAL/HAL_GPIO/hal_gpio_test.h]: File deleted as it´s duplicated, the correct one is in firmware/Sources /Test/HAL/

File: firmware/Sources/HAL/HAL_GPIO/hal_gpio_test.c (hunk 2 above)
Comment (RobertoAldea) [reply firmware/Sources/HAL/HAL_GPIO/hal_gpio_test.c]: File deleted as it´s duplicated, the correct one is in firmware/Sources /Test/HAL/

File: firmware/Sources/HAL/HAL_GPIO/hal_gpio.c (hunk 1 above)
Comment (RobertoAldea) [reply firmware/Sources/HAL/HAL_GPIO/hal_gpio.c]: Yes

File: firmware/Sources/Test/HAL/hal_gpio_test.h (hunk 5 above)
Comment (RobertoAldea) [reply firmware/Sources/Test/HAL/hal_gpio_test.h]: Yes

File: firmware/Sources/Test/HAL/hal_gpio_test.c (hunk 4 above)
Comment (RobertoAldea) [reply firmware/Sources/Test/HAL/hal_gpio_test.c]: Yes

**Review by pastorpflores — APPROVED** [review]

**Pull request merged**